code/game/gamemodes/nuclear/pinpointer.dm

```diff
@@ -8,290 +10,162 @@
-		visible_message("<span class='notice'>Authentication Disk Locator mode actived.</span>")
+	var/turf/here = get_turf(src)
+	var/turf/there = get_turf(target)
+	if(here.z != there.z)
+		icon_state = "pinon[nuke_warning ? "alert" : ""]null"
 		return
-	if(!home)
-		home = SSshuttle.getShuttle("syndicate")
-		if(!home)
-			icon_state = "pinonnull"
-			return
-	if(loc.z != home.z)	//If you are on a different z-level from the shuttle
-		icon_state = "pinonnull"
+	if(here == there)
+		icon_state = "pinon[nuke_warning ? "alert" : ""]direct"
 	else
-		setDir(get_dir(src, home))
-		switch(get_dist(src, home))
-			if(0)
-				icon_state = "pinondirect"
+		setDir(get_dir(here, there))
+		switch(get_dist(here, there))
 			if(1 to 8)
-				icon_state = "pinonclose"
+				icon_state = "pinon[nuke_warning ? "alert" : "close"]"
 			if(9 to 16)
-				icon_state = "pinonmedium"
+				icon_state = "pinon[nuke_warning ? "alert" : "medium"]"
 			if(16 to INFINITY)
-				icon_state = "pinonfar"
+				icon_state = "pinon[nuke_warning ? "alert" : "far"]"
 
-	spawn(5)
-		.()
-
-/obj/item/weapon/pinpointer/operative
-	name = "operative pinpointer"
-	icon = 'icons/obj/device.dmi'
-	desc = "A pinpointer that leads to the first Syndicate operative detected."
-	var/mob/living/carbon/nearest_op = null
-
-/obj/item/weapon/pinpointer/operative/attack_self()
-	if(!usr.mind || !(usr.mind in ticker.mode.syndicates))
-		usr << "<span class='danger'>AUTHENTICATION FAILURE. ACCESS DENIED.</span>"
-		return 0
-	if(!active)
-		active = 1
-		workop()
-		usr << "<span class='notice'>You activate the pinpointer.</span>"
-	else
-		active = 0
-		icon_state = "pinoff"
-		usr << "<span class='notice'>You deactivate the pinpointer.</span>"
-
-/obj/item/weapon/pinpointer/operative/proc/scan_for_ops()
-	if(active)
-		nearest_op = null //Resets nearest_op every time it scans
-		var/closest_distance = 1000
-		for(var/mob/living/carbon/M in mob_list)
-			if(M.mind && (M.mind in ticker.mode.syndicates))
-				if(get_dist(M, get_turf(src)) < closest_distance) //Actually points toward the nearest op, instead of a random one like it used to
-					nearest_op = M
-
-/obj/item/weapon/pinpointer/operative/proc/workop()
-	if(active)
-		scan_for_ops()
-		point_at(nearest_op, 0)
-		spawn(5)
-			.()
-	else
-		return 0
-
-/obj/item/weapon/pinpointer/operative/examine(mob/user)
-	..()
-	if(active)
-		if(nearest_op)
-			user << "Nearest operative detected is <i>[nearest_op.real_name].</i>"
-		else
-			user << "No operatives detected within scanning range."
+/obj/item/weapon/pinpointer/proc/my_god_jc_a_bomb() //If we should get the hell back to the ship
+	for(var/obj/machinery/nuclearbomb/bomb in machines)
+		if(bomb.timing)
+			if(!nuke_warning)
+				nuke_warning = TRUE
+				playsound(src, 'sound/items/Nuke_toy_lowpower.ogg', 50, 0)
+				if(isliving(loc))
+					var/mob/living/L = loc
+					L << "<span class='userdanger'>Your [name] vibrates and lets out a tinny alarm. Uh oh.</span>"

```

Comment:
Nah, that's intentional. If you listen to that sound, it's really low-quality. I'll upload it in a sec.